### PR TITLE
Test Case 4

### PR DESCRIPTION
**Pull request by vladstrazhnikov**

### CODE REVIEW

File: cypress/integration/TestCase4.spec.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+        cy.log('Click on Find a store link')
+        cy.xpath(mainPage.findStoreButton).click()
+        cy.url().should('include', '/store-locator')
+
+        cy.log('Filter Drive-thru option')
+        cy.get(storeLocatorPage.filterButton).click()
+        cy.get(storeLocatorPage.driveThruItem).click()
+        cy.get(storeLocatorPage.applyButton).click()
+        cy.get(storeLocatorPage.filterButton).contains('(1)')
```
Comment (KirylParkhomenka) [inline cypress/integration/TestCase4.spec.js:20]: lines 16-20 - you can create a separate method called `applySingleFilter(filter)`

File: cypress/pages/StoreLocatorPage.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
+export class StoreLocatorPage {
+    searchTermInput = '[data-e2e=searchTermInput]'
+    filterButton = '[data-e2e=filterButton]'
+    driveThruItem = '[data-e2e=locatorFilterItem--DT]'
+    applyButton = '[data-e2e=applyButton]'
+    locationList = '[data-e2e=locationList]>.base___3LiS9'
```
Comment (KirylParkhomenka) [inline cypress/pages/StoreLocatorPage.js:6]: `locationList` locator should be updated. Don't use this class

**Review by KirylParkhomenka — CHANGES_REQUESTED** [review]

File: cypress/integration/TestCase4.spec.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+describe('Test Case 4', () => {
+    const mainPage = new MainPage()
+    const storeLocatorPage = new StoreLocatorPage()
+
+    it('As a non-signed-in user I’m able to find 5 ‘Drive-Thru’ stores in Lockwood, MT, USA', () => {
+        cy.visit(urls.baseUrl)
+
+        cy.log('Click on Find a store link')
+        cy.xpath(mainPage.findStoreButton).click()
+        cy.url().should('include', '/store-locator')
```
Comment (KirylParkhomenka) [inline cypress/integration/TestCase4.spec.js:14]: In your previous PR you created `clickButton()` method.
Why not to reuse it here?

File: cypress/pages/StoreLocatorPage.js [hunk 4]
```diff
@@ -1,4 +1,20 @@
...
+    filterButton = '[data-e2e=filterButton]'
+    driveThruItem = '[data-e2e=locatorFilterItem--DT]'
+    applyButton = '[data-e2e=applyButton]'
+    locationList = '[data-e2e="locationList"]>article'
+
+    applySingleFilter(option, result) {
+        cy.get(this.filterButton).click()
+        cy.get(option).click()
+        cy.get(this.applyButton).click()
+        cy.get(this.filterButton).contains(result)
```
Comment (KirylParkhomenka) [inline cypress/pages/StoreLocatorPage.js:13]: Need to move assertion in test (best practice is to have assertion inside test.)

File: cypress/pages/StoreLocatorPage.js [hunk 5]
```diff
@@ -1,4 +1,20 @@
...
+    }
+
+    findStore(input, result) {
+        cy.get(this.searchTermInput).type(input)
+        cy.get(this.locationList).its('length').should('eq', result)
```
Comment (KirylParkhomenka) [inline cypress/pages/StoreLocatorPage.js:18]: Need to move assertion in test

**Review by KirylParkhomenka — CHANGES_REQUESTED** [review]

File: cypress/pages/StoreLocatorPage.js (hunk 5 above)
Comment (vladstrazhnikov) [reply cypress/pages/StoreLocatorPage.js:18]: I forgot that assertions should be in test files, I guess it is pointless to make findStore method then
so I moved line 17 back to test file then

**Review by KirylParkhomenka — APPROVED** [review]

**Pull request merged**